src/inc/CombFilter.h

```diff
@@ -0,0 +1,34 @@
+#if !defined(__CombFilter_hdr__)
+#define __CombFilter_hdr__
+
+#include "ErrorDef.h"
+#include "RingBuffer.h"
+
+class CCombFilterBase {
+public:
+    CCombFilterBase(float maxDelay, float sampleRate);
+    Error_t setGain(float gain);
+    virtual Error_t setDelay(float delay);
+    float getGain();
+    float getDelay();
+    virtual Error_t process(float *input, float *output, int numFrames) = 0;
+protected:
+    float sampleRate;
+    RingBuffer<float> delayline;
+    float gain = 0.5;
+};
+
+class FIRFilter : public CCombFilterBase {
+public:
+    using CCombFilterBase::CCombFilterBase;
```

Comment:
Compiler does not like "semantic issue"
<img width="811" alt="Screen Shot 2022-02-16 at 7 17 34 PM" src="https://user-images.githubusercontent.com/69863091/154379849-c65fa466-11a2-417a-bfb5-4ce864c50bab.png">
:

Reply:
C++11